Dockerfile

```diff
@@ -1,22 +1,24 @@
-FROM python:3.8-slim
-
-LABEL maintainer="info@datascience.ch"
+FROM python:3.8-alpine as base
+RUN apk add --no-cache curl tini && \
+    adduser -u 1000 -g 1000 -D kyaku
+WORKDIR /home/kyaku/renku-notebooks
 
-RUN pip install --no-cache-dir --disable-pip-version-check -U pip poetry && \
-    groupadd -g 1000 kyaku && \
-    useradd -u 1000 -g kyaku -m kyaku
+FROM base as builder
+ENV POETRY_HOME=/opt/poetry
+ENV POETRY_VIRTUALENVS_IN_PROJECT=true
+ENV POETRY_VIRTUALENVS_OPTIONS_NO_PIP=true
+ENV POETRY_VIRTUALENVS_OPTIONS_NO_SETUPTOOLS=true
+COPY poetry.lock pyproject.toml ./
+RUN apk add --no-cache alpine-sdk libffi-dev && \
+    mkdir -p /opt/poetry && \
+    curl -sSL https://install.python-poetry.org | python3 - && \
```

Comment:
I'm a bit worried that this could break things with suddenly changing the version out from under us. Should we pin this somehow?

Reply:
Yeah good point. Pinned it to 1.2.1.